source/js/buyers-guide/template-js-handler/product-page-comment-handler.js

```diff
@@ -4,34 +4,48 @@
  * we have this file which will push a datalayer event whenever a user submits a comment.
  */
 export default () => {
+  // Commento's parent div, in which it loads all of its content through JS.
   const commentoContainer = document.querySelector(
     "#view-product-page #commento"
   );
-  const submitButton = document.querySelector("#commento-submit-button-root");
 
-  if (commentoContainer && submitButton) {
-    submitButton.addEventListener(`click`, () => {
-      commentSubmitted();
+  const commentListUpdateHandler = (mutations) => {
+    let commentList = document.querySelector("#commento-main-area");
```

Comment:
the reason for defining this variable here, is because this div gets deleted/re-rendered during the following scenarios, meaning the observer would then have nothing to point at and would no longer push the datalayer event:
- If the user is not logged in and enters a comment, they get prompted to log in, after logging in this div gets re-rendered and the existing comment gets immediately submitted.
- If a user is not logged in and logs in, the div gets re-rendered.